churaverse-plugins-client/src/churarenPlugin/churarenPlayerPlugin/churarenPlayerPlugin.ts

```diff
@@ -0,0 +1,294 @@
+    this.ghostModeIndicatorUi.ghostModeIcon.deactivate()
+    this.socketController?.registerMessageListener()
+    this.keyboardController?.getStores()
+    this.keyboardController?.registerKeyActionListener()
+    this.setupKeyAction()
+  }
+
+  protected handleGameTermination(): void {
+    this.keyboardController?.unregisterKeyActionListener()
+    this.resetKeyAction()
+    this.socketController?.unregisterMessageListener()
+    this.playerItemStore.materialItemBoxContainer.remove()
+    this.churarenPlayerStore.ghostPlayerListUi.remove()
+    this.clearPlayerItemBox()
+    this.showAllGhostPlayers()
+    this.ghostModeIndicatorUi?.ghostModeIcon.deactivate()
+    resetChurarenPlayersStore(this.store)
+    resetPlayerItemStore(this.store)
+  }
+
+  protected handleMidwayParticipant(): void {
+    this.unsubscribeGameEvent()
+  }
+
+  private readonly onChurarenResult = (): void => {
+    this.hideItemBox()
+    this.clearPlayerItemBox()
+    this.ghostModeIndicatorUi?.ghostModeIcon.deactivate()
+    this.churarenPlayerStore.ghostPlayerListUi.updateGhostPlayerList([])
+    this.showAllGhostPlayers()
+  }
+
+  private readonly changeGhostMode = (ev: PlayerRespawnEvent): void => {
+    if (!this.isActive) return
+    // プレイヤーの透明化処理
+    const player = this.playerPluginStore.players.get(ev.id)
+    if (player === undefined) return
+
+    this.clearPlayerItemBox(player.id)
+    this.churarenPlayerStore.ghostModePlayerRepository.set(player.id, player)
+    if (this.playerPluginStore.ownPlayerId === player.id) {
+      this.ghostModeIndicatorUi?.ghostModeIcon.activate()
+      this.hideItemBox()
+    }
+    this.changeGhostPlayer(player.id)
+    this.updateGhostPlayerList()
+  }
+
+  private readonly onInvicibleTime = (ev: InvicibleTimeEvent): void => {
+    const playerId = ev.id
+    const invicibleTime = ev.invicibleTime
+    this.playerPluginStore.playerRenderers.get(playerId)?.blinkPlayer(invicibleTime)
+  }
+
+  private readonly getItem = (ev: GetChurarenItemEvent): void => {
+    const item = this.itemPluginStore.items.get(ev.itemId)
+    const renderer = this.itemPluginStore.itemRenderers.get(ev.itemId)
+
+    // TODO: 錬金アイテムを持っている場合も考慮する
+    if (item === undefined || renderer === undefined) return
+
+    this.playerItemStore.materialItems.set(ev.playerId, item)
+    this.playerItemStore.materialItemRenderers.set(ev.itemId, renderer)
+    this.itemPluginStore.items.delete(ev.itemId)
+    this.itemPluginStore.itemRenderers.delete(ev.itemId)
+    if (ev.playerId === this.playerPluginStore.ownPlayerId) {
+      this.updateMaterialItemBox(ev.playerId)
+    }
+  }
+
+  private readonly onPlayerWalk = (ev: PlayerWalkEvent): void => {
+    const player = this.playerPluginStore.players.get(ev.id)
+    if (player === undefined) return
+    const speed = ev.speed ?? GRID_SIZE / GRID_WALK_DURATION_MS
+
+    // TODO: 錬金アイテムを持っている場合の処理を追加
+    const itemBoxes = this.playerItemStore.materialItems.getAllItem(ev.id)
```

Comment:
itemBox(=アイテムの入れ物) と アイテムの集合 は厳密には意味が異なります
この処理の場合、各アイテムに対してプレイヤーの座標に向けて移動させる処理が主体なので、対象はアイテム群になります。
このことから、命名はitemsとかでいい気がしますね
```suggestion
    const items = this.playerItemStore.materialItems.getAllItem(ev.id)
```